plugins/aea-ledger-hwi/aea_ledger_hwi/account.py

```diff
@@ -0,0 +1,408 @@
+    """Hardware wallet signed transaction"""
+
+    v: int
+    r: int
+    s: int
+
+
+class UnsignedDynamicTransaction(HashableRLP):
+    """Unsigned dynamic transaction."""
+
+    fields = [
+        ("chain_id", rlp.sedes.big_endian_int),
+        ("nonce", rlp.sedes.big_endian_int),
+        ("max_priority_fee_per_gas", rlp.sedes.big_endian_int),
+        ("max_fee_per_gas", rlp.sedes.big_endian_int),
+        ("gas_limit", rlp.sedes.big_endian_int),
+        ("destination", address),
+        ("amount", rlp.sedes.big_endian_int),
+        ("data", rlp.sedes.binary),
+        ("access_list", access_list_sede_type),
+    ]
+
+
+class SignTransactionAPDU:
+    """Sign transaction APDU codes"""
+
+    INS = 0x04
+    P1_0 = 0x00
+    P1 = 0x80
+    P2 = 0x00
+
+
+class GetAccountAPDU:
+    """Get account APDU codes"""
+
+    INS = 0x02
+    P1 = 0x01
+    P2 = 0x01
+
+
+SignedTransactionStruct = Struct(
+    v=BytesInteger(1),
+    r=BytesInteger(32),
+    s=BytesInteger(32),
+)
+
+
+class HWIErrorCodes:
+    """HWI com errors."""
+
+    DEVICE_LOCKED = 21781
+    EHTEREUM_APP_NOT_OPEN = 25870
+
+
+@contextmanager
+def reraise_from_hwi_com_error() -> Generator:
+    """Reraise ledger communication exception as `HWIError`"""
+
+    try:
+        yield
+    except CommException as e:
+        if e.sw == HWIErrorCodes.DEVICE_LOCKED:
+            raise HWIError(message="Device is locked", sw=e.sw)
+        elif e.sw == HWIErrorCodes.EHTEREUM_APP_NOT_OPEN:
+            raise HWIError(message="Please open ethereum app in your device", sw=e.sw)
+        else:
+            raise HWIError(message=e.message, sw=e.sw, data=e.data)
+
+
+class HWIAccount:
+    """Hardware wallet interface as ethereum account similar to `eth_account.Account` to represent `Crypto.entity`"""
+
+    default_account: Optional[HWIAccountData]
+
+    def __init__(
+        self,
+        default_device: int = 0,
+        default_key_index: int = 0,
+    ) -> None:
+        """Initialize object."""
+
+        self.default_device = default_device
+        self.default_key_index = default_key_index
+        self.default_account = None
+
+    @property
+    def devices(self) -> List[Device]:
+        """Returns the list of available devices."""
+        devices = enumerate_devices()
+        if len(devices) == 0:
+            raise HWIError(message="Cannot find any ledger device", sw=0)
+        return devices
+
+    def get_client(
+        self,
+        device_index: Optional[int] = None,
+    ) -> LedgerClient:
+        """Get ledger client."""
+        device_index = device_index or self.default_device
+        return LedgerClient(device=self.devices[device_index])
+
+    def get_account(
+        self,
+        key_index: Optional[int] = None,
+        device_index: Optional[int] = None,
+    ) -> HWIAccountData:
+        """Get hardware wallet account."""
+
+        key_index = key_index or self.default_key_index
+        path = m / h(44) / h(60) / h(key_index) / 0 / 0
```

Comment:
Please re-checking this 
m / purpose' / coin_type' / account' / change / address_index
path = m / h(44) / h(60) / h(key_index) / 0 / 0
The situation ledger looks rather confusing. 
And as I remember, the same metamask has a separate (non-standard) path. 
I'm not saying that it's wrong - just need to double-check.
https://ethereum.stackexchange.com/questions/19055/what-is-the-difference-between-m-44-60-0-0-and-m-44-60-0
https://github.com/MetaMask/metamask-extension/blob/99ed42b3dc4e1cf7fb427f433ff7cb5bda732a4f/ui/pages/create-account/connect-hardware/index.js#L25-L27
https://github.com/MyCryptoHQ/MyCrypto/issues/2070

Reply:
I took this path format from the README file of the repo https://github.com/LedgerHQ/apduboy/blob/master/README.md?plain=1#L74

Reply:
OK. Probably good!
I would recommend a test with some non-obvious account (for example: account #3) and compare the results for this library, ledger live and metamask.
Off topic question
"apduboy is currently in alpha. Please do NOT use this for signing real transactions." (c)
by [https://github.com/LedgerHQ/apduboy/blob/master/README.md]
Is it just a standard phrase or is there really some kind of problem?